README.md

```diff
@@ -67,6 +67,25 @@ The following attributes are available for all models of `viamrtsp` cameras:
 
 To test your camera, go to the [**CONTROL** tab](https://docs.viam.com/fleet/control/) of your machine in the [Viam app](https://app.viam.com) and expand the camera's panel.
 
+## RTSP stream discovery
+### Use `DiscoverComponents`
+In addition to being able to manually specify RTSP addresses to stream from, `viamrtsp` also offers a discovery utility to search for IP cameras connected to your LAN and return their respective RTSP addresses.
+
+Each model in `viamrtsp` registers a `Discover` method that can be invoked via a `DiscoverComponents` robot API call. See [examples/discovery-client](examples/discovery-client/client.go) for an example usage of discovery and [`DiscoverComponents`](https://docs.viam.com/appendix/apis/robot/#discovercomponents) for documentation.
+
+As of time of writing, Viam supports `DiscoverComponents` in Flutter and Golang.
+
+### Common RTSP discovery pitfalls
+#### DHCP
+IP camera does not support DHCP, and does not have an assigned IP after connecting to your LAN. In this case, you'll have to assign the camera's IP manually. This can be done over your network provider's UI.
+Some network provider UIs also allow you to find a camera using its MAC address or the specific Ethernet port it's connected to, and manually assign an IP address from there.
+
+#### ONVIF adherence
+Discovery relies on the IP camera adhering to standard ONVIF methods such as getting device metadata, media profiles, and stream URIs. It will not work with non-existent or incompatible ONVIF camera integrations.
```

Comment:
Should we be more descriptive here on which Onvif Profile level needs to be satisfied.

Pretty sure it is the baseline `Profile S`.

Reply:
Got it made edit

Reply:
Damn we should change tests to not run on README changes lol